examples/progress.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 System76 <info@system76.com>
+// SPDX-License-Identifier: MIT or Apache-2.0
+
+//! A component which allows the caller to define what options ae in its list.
+//!
+//! On init of the view, an output is sent to the caller to request to load widgets.
+//!
+//! Clicking a button will open the webpage to that option.
+//!
+//! Clicking the clear button will clear the list box and send a command to the
+//! background that waits 2 seconds before issuing a reload command back to the
+//! component, which forwards the reload command back to the caller of the
+//! component, which then issues to reload the widgets again.
+
+use futures::FutureExt;
+use gtk::prelude::*;
+use relm4::*;
+
+fn main() {
+    gtk::Application::builder()
+        .application_id("org.relm4.ProgressExample")
+        .launch(|_app, window| {
```

Comment:
I prefer using `RelmApp` for startup as it reduces the boilerplate code. Yet of course `ApplicationBuilder` is fine, too.